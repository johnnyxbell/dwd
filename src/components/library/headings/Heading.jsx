import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/macro';
import { WHITE } from '../../../styles/colors';
import { WEIGHT, FONT_FAMILY } from '../../../styles/typography';

const Title = styled.h1`
  font-size: 1.5rem;
  color: ${WHITE};
  font-weight: ${WEIGHT.THIN};
  font-family: ${FONT_FAMILY};
`;

const Heading = ({ title }) => {
  return <Title>{title}</Title>;
};

Heading.propTypes = {
  title: PropTypes.string
};

export default Heading;
