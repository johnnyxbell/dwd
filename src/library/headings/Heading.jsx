import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/macro';
import { SILVER } from '../../styles/colors';
import { WEIGHT, FONT_FAMILY } from '../../styles/typography';

const Title = styled.h1`
  font-size: 1.5rem;
  color: ${SILVER};
  font-weight: ${WEIGHT.THIN};
  font-family: ${FONT_FAMILY};
`;

const TrailingText = styled.p`
  font-size: 1rem;
  color: ${SILVER};
  font-weight: ${WEIGHT.THIN};
  font-family: ${FONT_FAMILY};
`;

const Heading = ({ title, trail }) => {
  return (
    <div>
      <Title>{title}</Title>
      {trail && <TrailingText>{trail}</TrailingText>}
    </div>
  );
};

Heading.propTypes = {
  title: PropTypes.string
};

export default Heading;
