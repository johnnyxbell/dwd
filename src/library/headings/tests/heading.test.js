import React from 'react';
import renderer from 'react-test-renderer';
import Heading from '../Heading';

describe('Heading', () => {
  test('Heading', () => {
    expect(renderer.create(<Heading title="hello" />).toJSON()).toMatchSnapshot();
  });
  test('Heading with trailing text', () => {
    expect(renderer.create(<Heading title="hello" trail="I am some trailing text" />).toJSON()).toMatchSnapshot();
  });
});
