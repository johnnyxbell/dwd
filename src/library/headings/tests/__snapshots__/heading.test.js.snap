// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Heading 1`] = `
<div>
  <h1
    className="Heading__Title-sc-2qqen3-0 dNmNsH"
  >
    hello
  </h1>
</div>
`;

exports[`Heading Heading with trailing text 1`] = `
<div>
  <h1
    className="Heading__Title-sc-2qqen3-0 dNmNsH"
  >
    hello
  </h1>
  <p
    className="Heading__TrailingText-sc-2qqen3-1 iSBTRn"
  >
    I am some trailing text
  </p>
</div>
`;
